pkg/agent/resourcemanager/fetcher/kubelet/topology/util.go

```diff
@@ -18,108 +18,114 @@ package topology
 
 import (
 	"sort"
-
-	v1 "k8s.io/api/core/v1"
-	"k8s.io/klog/v2"
+	"strconv"
 
 	info "github.com/google/cadvisor/info/v1"
-
 	nodev1alpha1 "github.com/kubewharf/katalyst-api/pkg/apis/node/v1alpha1"
+	"github.com/kubewharf/katalyst-core/pkg/util"
 )
 
-// GenerateSocketStatus generate a map of socket id to socket status,
-// which includes numa id and numa capacity of each socket.
-func GenerateSocketStatus(
-	numaCapacity map[int]*v1.ResourceList,
-	numaAllocatable map[int]*v1.ResourceList,
-	numa2socket map[int]int,
-) map[int]*nodev1alpha1.SocketStatus {
-	socketStatusMap := make(map[int]*nodev1alpha1.SocketStatus)
-
-	// build numa status and set NumaID, Allocatable, Capacity
-	for numaID, capacity := range numaCapacity {
-		socketID, ok := numa2socket[numaID]
-		if !ok {
-			klog.Errorf("[GenerateSocketStatus] not found socketID for node %d", numaID)
-			continue
+type ZoneAttributes []nodev1alpha1.Attribute
+
+type ZoneTopology map[util.ZoneKey]*ZoneTopology
+
+type ZoneAllocations []*nodev1alpha1.Allocation
+
+// GenerateTopologyZoneStatus generates topology zone status
+func GenerateTopologyZoneStatus(
+	zoneTopology *ZoneTopology,
+	allocationsMap map[util.ZoneKey]ZoneAllocations,
+	resourcesMap map[util.ZoneKey]nodev1alpha1.Resources,
+	attributesMap map[util.ZoneKey]ZoneAttributes,
+) []*nodev1alpha1.TopologyZone {
+	if zoneTopology == nil {
+		return nil
+	}
+
+	var result []*nodev1alpha1.TopologyZone
+	for zone, topology := range *zoneTopology {
+		topologyZone := &nodev1alpha1.TopologyZone{
+			Type: zone.Meta.Type,
+			Name: zone.Meta.Name,
 		}
 
-		socket, ok := socketStatusMap[socketID]
-		if !ok {
-			socketStatusMap[socketID] = &nodev1alpha1.SocketStatus{
-				SocketID: socketID,
-			}
-			socket = socketStatusMap[socketID]
+		if resources, ok := resourcesMap[zone]; ok {
+			topologyZone.Resources = resources
+		}
+
+		if attributes, ok := attributesMap[zone]; ok {
+			topologyZone.Attributes = attributes
 		}
 
```

Comment:
move this implementation move flexible, meaning that when we want to add a new children, we only need to add a specific function without adjusting the framework or skeleton